llvm/lib/Target/ARM/MCTargetDesc/ARMInstPrinter.cpp

```diff
@@ -1264,8 +1277,8 @@ void ARMInstPrinter::printT2AddrModeImm0_1020s4Operand(
   O << markup("<mem:") << "[";
   printRegName(O, MO1.getReg());
   if (MO2.getImm()) {
-    O << ", " << markup("<imm:") << "#" << formatImm(MO2.getImm() * 4)
-      << markup(">");
+    O << ", " << markup(MO1.getReg() == ARM::PC ? "<pcrel:" : "<imm:")
```

Comment:
This code is repeated a lot, factor it out.

Reply:
I'm not 100% sure there is a good way to do that. Consider that they are subtly different.

As such the `IsSub` thing is better left alone. The best I could do is something like
```cpp
std::string immediateOffsetMarkup(const MCOperand &Operand) {
  return markup(MO1.getReg() == ARM::PC ? "<pcrel:" : "<imm:");
}
```
but considering that's just a line, I'm not sure it's really worth an extra function.

Reply:
I'd like that function, just to avoid copy paste/refactoring mistakes now or in the future.

Reply:
Done.